Signal/src/view controllers/CallViewController.swift

```diff
@@ -205,6 +226,191 @@ class CallViewController: UIViewController {
+        ongoingCallRows.last!.autoPinEdge(toSuperviewEdge:.bottom)
+
+        // Incoming call controls
+        acceptIncomingButton = createButton(imageName:"call",
+                                            action:#selector(didPressAnswerCall),
+                                            buttonSize:buttonSize)
+        declineIncomingButton = createButton(imageName:"endcall",
+                                             action:#selector(didPressDeclineCall),
+                                             buttonSize:buttonSize)
+
+        incomingCallControlsRow = rowWithSubviews(subviews:[acceptIncomingButton, declineIncomingButton],
+                                                  fixedHeight:buttonSize)
+        self.view.addSubview(incomingCallControlsRow)
+    }
+
+    func createButton(imageName : String!, action : Selector!, buttonSize : CGFloat!) -> UIButton {
+        let image = UIImage(named:imageName)
+        Logger.error("button \(imageName) \(NSStringFromCGSize(image!.size))")
+        Logger.flush()
+        let button = UIButton()
+        button.setImage(image, for:.normal)
+        button.addTarget(self, action:action, for:.touchUpInside)
+        button.autoSetDimension(.width, toSize:buttonSize)
+        button.autoSetDimension(.height, toSize:buttonSize)
+        return button
+    }
+
+    // Creates a row view that evenly spaces its subviews horizontally.
+    // If there is only a single subview, it is centered.
+    func rowWithSubviews(subviews : Array<UIButton>, fixedHeight : CGFloat) -> UIView {
+        let row = UIView()
+        row.setContentHuggingVerticalHigh()
+        row.autoSetDimension(.height, toSize:fixedHeight)
+        
+        if subviews.count > 1 {
+            var lastSubview : UIView?
+            var lastSpacer : UIView?
+            for subview in subviews {
+                row.addSubview(subview)
+                subview.setContentHuggingHorizontalHigh()
+                subview.autoVCenterInSuperview()
+
+                if lastSubview != nil {
+                    let spacer = UIView()
+                    spacer.isHidden = true
+                    row.addSubview(spacer)
+                    spacer.autoPinEdge(.left, to:.right, of:lastSubview!)
+                    spacer.autoPinEdge(.right, to:.left, of:subview)
+                    spacer.setContentHuggingHorizontalLow()
+                    spacer.autoVCenterInSuperview()
+                    if lastSpacer != nil {
+                        spacer.autoMatch(_:.width, to:.width, of:lastSpacer!)
+                    }
+                    lastSpacer = spacer
+                }
+
+                lastSubview = subview
+            }
+            subviews.first!.autoPinEdge(toSuperviewEdge:.left)
+            subviews.last!.autoPinEdge(toSuperviewEdge:.right)
+
+            Logger.error("row \(subviews.count) -> \(row.subviews.count)")
+
+        } else if subviews.count == 1 {
```

Comment:
I forgot what condition the else was from. How about moving this up to an early return?